code/modules/food_and_drinks/machinery/dinner_button.dm

```diff
@@ -0,0 +1,25 @@
+/obj/machinery/foodbutton
+	name = "dinner button"
+	desc = "Casts an announcement to the station, reminding the crew about food."
+	icon = 'icons/obj/service/kitchen.dmi'
+	icon_state = "button_supper"
+	layer = OBJ_LAYER
+	anchored = TRUE
+	pass_flags = PASSTABLE // Able to place on tables
+	var/button_used = FALSE // only one announcement per shift
+	var/button_sound = 'sound/machines/pda_button2.ogg'
+
+/obj/machinery/foodbutton/Initialize(mapload)
+	. = ..()
+
+/obj/machinery/foodbutton/attack_hand(mob/living/user)
+	. = ..()
+	announce()
+
+/obj/machinery/foodbutton/proc/announce(mob/living/user)
+	if(button_used)
+		return FALSE
+	minor_announce("Attention crew! It's lunchtime! Head over for a delicious meal prepared just for you. Bon appétit!", "Kitchen announcement")
+	playsound(src, button_sound, 70, TRUE, -1)
+	button_used = TRUE
+	return TRUE
```

Comment:
Why are you returning TRUE or FALSE for a custom proc that's not used in an if statement